data/core/macros/amla.cfg

```diff
@@ -12,19 +31,283 @@
+                                    apply_to=attack
+                                    increase_damage=4
+                                    range=ranged
+                                [/effect]
+                                {AMLA_EFFECTS_HEAL_AND_RAISE_XP}
+                            [/advancement]
+
+                            # +2 ranged damage
+                            [advancement]
+                                id=ranged_increase2
+                                description= _ "Gain +2 ranged damage."
+                                strict_amla=yes
+                                max_times=-1
+                                [filter]
+                                    [has_attack]
+                                        range=ranged
+                                        number=2
+                                    [/has_attack]
+                                    [not]
+                                        [has_attack]
+                                            range=ranged
+                                            number=3-99
+                                        [/has_attack]
+                                    [/not]
+                                [/filter]
+                                [effect]
+                                    apply_to=attack
+                                    increase_damage=2
+                                    range=ranged
+                                [/effect]
+                                {AMLA_EFFECTS_HEAL_AND_RAISE_XP}
+                            [/advancement]
+
+                            # +1 ranged damage
+                            [advancement]
+                                id=ranged_increase3
+                                description= _ "Gain +1 ranged damage."
+                                strict_amla=yes
+                                max_times=-1
+                                [filter]
+                                    [has_attack]
+                                        range=ranged
+                                        number=3-99
+                                    [/has_attack]
+                                [/filter]
+                                [effect]
+                                    apply_to=attack
+                                    increase_damage=1
+                                    range=ranged
+                                [/effect]
+                                {AMLA_EFFECTS_HEAL_AND_RAISE_XP}
+                            [/advancement]
+                        [/effect]
+                    [/object]
+                [/modify_unit]
+
+                #--------------------
+                # RESET AMLAS
+                #--------------------
+                [event]
+                    name=post advance
+                    [clear_variable]
+                        name=amla_currently_advancing_id
+                    [/clear_variable]
+                    [remove_object]
```

Comment:
Why it should be removed? Players will see wrong information on sidebar if amla is added only after max xp is reached.

Reply:
Good point. How does this look?

Reply:
From review the new design seems correct.